src/internal.h

```diff
@@ -34,6 +34,13 @@ inline pid_t gettid(void) {
   return syscall(__NR_gettid);
 }
 #endif
+#ifdef __APPLE__
+inline pid_t gettid(void) {
+  uint64_t tid;
+  pthread_threadid_np(NULL, &tid);
+  return tid;
```

Comment:
This chops off the first 32 bits of the thread ID & could result in ID collisions. Perhaps a better solution would be to define a `tid_t` that is 32 bits on Linux or 64 bits on macOS?

Reply:
@nornagon TBH, ideally we would just use the 64-bit ID everywhere, but I opted for 32-bits because this ID is stored in the `MiniHeap`, which has a hard size limit of 64-bytes and is pretty jam packed already.

http://elliotth.blogspot.com/2012/04/gettid-on-mac-os.html has some more info, and it doesn't seem like there is an easy way to get a 32-bit TID on macOS

Reply:
I would assume keeping the lower 32 bits would work fine _in practice_ (because they are probably assigned in consecutive order, and it seems pretty unlikely that an app would cycle through 2^32 threads in its lifetime).

Reply:
We could also mix up the bits to make this probabilistically safer.

@bobby-stripe Is there a downside to incorporating this?

Reply:
I think we should `return static_cast<uint32_t>(tid);` to make it clear we are truncating the bits -- assuming `pid_t` is 32-bit.